libsolidity/ast/Types.cpp

```diff
@@ -1027,10 +1027,10 @@ TypePointer RationalNumberType::unaryOperatorResult(Token _operator) const
 	default:
 		return TypePointer();
 	}
-	return make_shared<RationalNumberType>(value);
+	return TypeResult(make_shared<RationalNumberType>(value));
```

Comment:
In places like these, we could use the newly created `Success(...)` API to return the value **and** indicate success. What do you think?

Reply:
I will change that as soon as the free-standing functions are implement.

Reply:
@christianparpart I decided to make a cut and leave those changes for further improvements. Adding those would be really nice but also need some more code (unfortunately). For the time being you could write ``Result<bool>::Ok(true)`` or ``Result<bool>::Err()``.